mito-ai/mito_ai/handlers.py

```diff
@@ -181,13 +182,30 @@ async def on_message(self, message: str) -> None: # type: ignore
             )
             self.reply(reply)
             return
+        
         try:
-            
             # Get completion based on message type
             completion = None
+            message_id = parsed_message.get('message_id')
+            stream = parsed_message.get('stream')
+
             if type == MessageType.CHAT:
                 chat_metadata = ChatMessageMetadata(**metadata_dict)
-                completion = await get_chat_completion(chat_metadata, self._llm, message_history)
+                
+                # Handle streaming if requested and available
+                if stream and self._llm.can_stream:
+                    # Use stream_chat_completion to stream the response
+                    async for chunk in stream_chat_completion(
+                        chat_metadata, 
+                        self._llm, 
+                        message_history, 
+                        message_id,
+                    ):
+                        self.reply(chunk)                    
+                    return
```

Comment:
We have this `async for chunk in ... ` logic in three places: 

1. Handler
2. Completion Handler 
3. Provider

We send the chunks all the way back up 3 -> 2 -> 1 and then just fire off the replies when we get them back. 

I wonder if we could simplify this by just passing the `self.reply` function down to the provider and then fire off replies from there instead. Then the provider can also just return the accumulated result to the Completion Handler and we can save it in the message history